### PR TITLE
Allow snowflake config variable names to be set for user auth

### PR DESCRIPTION
**Pull request by naterush**
# Description

This greatly improves the snowflake demos I can give - I always spend time fiddling around with these.

# Testing

Test it!

# Documentation

No.

### CODE REVIEW

File: mitosheet/mitosheet/api/get_validate_snowflake_credentials.py [hunk 1]
```diff
@@ -25,9 +26,29 @@
 
 def get_cached_snowflake_credentials() -> Optional[SnowflakeCredentials]:
     global __cached_snowflake_credentials
+
+    # If the snowflake connector is importer, and the SNOWFLAKE_ACCOUNT environment variable is set,
+    # then we can use that to prepopulate the snowflake credentials.
+    if SNOWFLAKE_CONNECTOR_IMPORTED and 'SNOWFLAKE_ACCOUNT' in os.environ:
```
Comment (naterush) [inline mitosheet/mitosheet/api/get_validate_snowflake_credentials.py:32]: Should get for all of them.

**Review by naterush — COMMENTED** [review]
Gotta update Github action.

**Review by aarondr77 — COMMENTED** [review]
LGTM, but I think another great polish for demoing (another thing that comes up a lot for me when I show these features) is if there are env set, to use them in the generated code instead of exposing the username and password.

File: mitosheet/mitosheet/code_chunks/snowflake_import_code_chunk.py [hunk 2]
```diff
@@ -36,13 +38,26 @@ def get_description_comment(self) -> str:
...
+                'cur = con.cursor()',
+            ]
+        else:
+            imports.append('import os')
+            snowflake_connection_code = [
+                'con = snowflake.connector.connect(',
+                f'    user=os.environ[\'SNOWFLAKE_USERNAME\'],',
+                f'    password=os.environ[\'SNOWFLAKE_PASSWORD\'],',
+                f'    account=os.environ[\'SNOWFLAKE_ACCOUNT\'],',
+                ')',
```
Comment (aarondr77) [inline mitosheet/mitosheet/code_chunks/snowflake_import_code_chunk.py:57]: This should be the full connection string I think, not just the user, password, and account!

**Comment by aarondr77** [comment]
# Importing twice is broken
1. Import one table using Snowflake
2. Import the same query again (or a different one)
3. Get an error

**Comment by naterush** [comment]
Merging when tests pass

**Pull request merged**